docs/dev/framework/security.md

```diff
@@ -0,0 +1,275 @@
+```
+{{% /notice %}}
+
+
+## Custom Back End Access Rights
+
+To implement your own back end access rights (e.g. for custom modules in the back end) the following steps are necessary:
+
+1. Add your new permission to `$GLOBALS['TL_PERMISSIONS']`.
+2. Add your new permission to `tl_user` and `tl_user_group`.
+
+```php
+// contao/config/config.php
+$GLOBALS['TL_PERMISSIONS'][] = 'my_permission';
+```
+
+```php
+// contao/dca/tl_user.php
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+
+$GLOBALS['TL_DCA']['tl_user_group']['fields']['my_permission'] = [
+    'exclude' => true,
+    'inputType' => 'checkbox',
+    'sql' => ['type' => 'boolean', 'default' => false],
+];
+
+PaletteManipulator::create()
+    ->addLegend('my_legend', null)
+    ->addField('my_permission', 'my_legend', PaletteManipulator::POSITION_APPEND)
+    ->applyToPalette('extend', 'tl_user')
+    ->applyToPalette('custom', 'tl_user')
+;
+```
+
+```php
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+
+$GLOBALS['TL_DCA']['tl_user_group']['fields']['my_permission'] = [
+    'exclude' => true,
+    'inputType' => 'checkbox',
+    'sql' => ['type' => 'boolean', 'default' => false],
+];
+
+PaletteManipulator::create()
+    ->addLegend('my_legend', null)
+    ->addField('my_permission', 'my_legend', PaletteManipulator::POSITION_APPEND)
+    ->applyToPalette('default', 'tl_user_group')
+;
+```
+
+Once that is done you can check for this permission in your own back end controller for example. These permissions can be checked via the 
+security helper by using the `contao_user.*` attribute. Since our new permission is only a boolean field, we check whether it is `true`. We 
+also check that the current user is not `ROLE_ADMIN` because administrators should always be able to access the controller.
+
+```php
+// src/Controller/BackendController.php
+namespace App\Controller;
+
+use Contao\CoreBundle\Exception\AccessDeniedException;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Security;
+use Twig\Environment;
+
+/**
+ * @Route("/contao/my-backend-route",
+ *     name=BackendController::class,
+ *     defaults={"_scope": "backend"}
+ * )
+ */
+class BackendController
+{
+    private $twig;
+    private $security;
+
+    public function __construct(Environment $twig, Security $security)
+    {
+        $this->twig = $twig;
+        $this->security = $security;
+    }
+
+    public function __invoke(): Response
+    {
+        if (!$this->security->isGranted('ROLE_ADMIN') && !$this->security->isGranted('contao_user.my_permission', true)) {
+            throw new AccessDeniedException('Not enough permissions to access this controller.');
+        }
+
+        return new Response($this->twig->render('my_backend_route.html.twig', []));
+    }
+}
+```
+
+{{% notice note %}}
+We do not need to check whether a user is logged in this case, since the Contao firewall automatically checks this
```

Comment:
but `ROLE_ADMIN` does that check as well

Reply:
Yes, but this is a check for _not_ admin, which would be true for an unauthenticated user - but due to the firewall we don't have to check for that.